### DISCUSSION TITLE
Day 2 : Access the 'JavaScript' string characters using a random number.

### DISCUSSION THREAD

**Discussion by dilawar23n** [Asabeneh/30-Days-Of-JavaScript#642] [Q&A]
Hello, I'm stuck on this. Can anyone help

**Comment by Al-Ponce** [comment]
Hello there, have a look at the "String Methods" part in Day 2.
Let's access the fourth character in 'JavaScript' string for example.

    let string = 'JavaScript'
    let fourthLetter = string[3]
    console.log(fourthLetter) // a

or maybe just `string.charAt(random number)`

**Reply by KhuramShahzad-5335** [reply]
can u plz check my answer:

//-------------------------------10
/*
    Access the 'JavaScript' 
    string characters using a random number
*/

const accessStr = 'JavaScript'
lastIndex = accessStr.length - 1
console.log(lastIndex)
randomNumToAccess = Math.ceil(Math.random() * lastIndex)
console.log(randomNumToAccess)
console.log(accessStr.charAt(randomNumToAccess))

**Reply by Al-Ponce** [reply]
I think it works.
Maybe just the question made us a bit confusing.
You can type a random num yourself either.

    let string = 'JavaScript'
    console.log(string.charAt(random number))
    
    //random number = 0~9

**Reply by As-Yung** [reply]
Actually it is wrongly explained, there are three types of variables in JavaScript (var, let, const); var and let variables can change their values, const cannot since its use is to validate or declare global information.

ERRONEOUS example in the guide:

let word = "world"
word[2] = 'y'

Generates error --> this is true, however to make a change in the variable word in a string, it can be done by changing its content like this:

let word = "world"
word = "verydo"

This shows that primitive variables can be changed, since in JavaScript as in other languages what defines the type of a variable is not its content, but its definition (var, let, const).

**Comment by KhuramShahzad-5335** [comment]
> Hello, I'm stuck on this. Can anyone help

//-------------------------------10
/*
    Access the 'JavaScript' 
    string characters using a random number
*/

const accessStr = 'JavaScript'
lastIndex = accessStr.length - 1
console.log(lastIndex)
randomNumToAccess = Math.ceil(Math.random() * lastIndex)
console.log(randomNumToAccess)
console.log(accessStr.charAt(randomNumToAccess))

**Comment by kattasahan** [comment]
**My Solution**

const JS = "JavaScript";
console.log(JS[Math.floor(Math.random() * (JS.length - 0 + 1))]);

**Comment by haikalgakbar** [comment]
We can utilize Math.random() to generate random number and multiply it with String.length so that the resulting number matches the length of the String.

My answer is like this:
```
const str = 'JavaScript';

console.log(str[Math.floor(Math.random() * str.length)]);
```
